unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,235 @@
+        """Check whether if is active is false throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API registration process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background "
+                                 "and will notify you as soon as the "
+                                 "registration has been completed."):
+            redis_client_auto_register.register(is_blocking=True)
+        mock_wait_until_timeout_blocking.assert_called()
+
+    @staticmethod
+    def test_register_self_active_true_throws_exception(redis_client_auto_register):
+        """Check whether if is active is true throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API is already registered to the market."):
+            redis_client_auto_register.is_active = True
+            redis_client_auto_register.register(is_blocking=False)
+
+    @staticmethod
+    @patch("uuid.uuid4", return_value="some-transaction-uuid")
+    def test_unregister(mock_uuid, redis_client_auto_register):
+        """Check whether the redis.db.publish is called with correct arguments."""
+        redis_client_auto_register.is_active = True
+        redis_client_auto_register.unregister(is_blocking=False)
+        data = {"name": AREA_ID, "transaction_id": mock_uuid()}
+        redis_client_auto_register.redis_db.publish.assert_called_once_with(
+            f"{AREA_ID}/unregister_participant",
+            json.dumps(data))
+
+    @staticmethod
+    def test_unregister_is_active_false_throws_exception(redis_client_auto_register):
+        """Check if is active set to false throws exception when user is already unregistered."""
+        with pytest.raises(RedisAPIException,
+                           match="API is already unregistered from the market."):
+            redis_client_auto_register.is_active = False
+            redis_client_auto_register.unregister(is_blocking=True)
+
+    @staticmethod
+    @patch("d3a_api_client.redis_client_base.wait_until_timeout_blocking",
+           side_effect=AssertionError)
+    def test_unregister_is_active_true_throws_exception(
+            mock_wait_until_timeout_blocking,
+            redis_client_auto_register):
+        """Check if is active set to true throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API unregister process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background and "
+                                 "will notify you as soon as "
+                                 "the unregistration has been completed."):
+            redis_client_auto_register.is_active = True
+            redis_client_auto_register.unregister(is_blocking=True)
+        mock_wait_until_timeout_blocking.assert_called()
```

Comment:
Nit, having said the exception was raised, this is indeed called so this line can be removed

Reply:
It's true in the current case. However, the exception could be raised by any other call/line in the `unregister()` method, without `mock_wait_until_timeout_blocking` to be called. I think the current version makes it a bit clearer where exactly the exception was raised.
I agree that it's not strictly necessary to be so specific here, but I'd keep it here as it doesn't harm. Would this be ok for you?

Reply:
Sure, it's okay
It's just the the Redis exception will be thrown upon catching an AssertionError, which is the side effect specified in the mock_wait_until_timeout_blocking , this is why I thought it's somehow clear but indeed no harm in keeping it